// Copyright 2023 Ant Group Co., Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package secretflow.component;

import "secretflow/protos/component/comp.proto";
import "secretflow/protos/component/data.proto";

// Evaluate a node.
//   evaluate(NodeEvalParam param, SFClusterConfig sf_config) -> NodeEvalResult
// NodeEvalParam contains all the information to evaluate a component.
message NodeEvalParam {
  // domain of component
  string domain = 1;

  // name of component
  string name = 2;

  // version of component
  string version = 3;

  // The attribute value.
  repeated string attr_paths = 4;

  repeated Attribute attrs = 5;

  // The input values
  // NOTE: name of DistData doesn't need to match name of input in Comp
  // definition.
  repeated DistData inputs = 6;

  repeated string output_uris = 7;
}

message NodeEvalResult {
  // Output values.
  repeated DistData outputs = 1;
}
